velero-plugins/imagestreamtag/restore.go

```diff
@@ -30,94 +30,101 @@ func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
+				p.Log.Info(fmt.Sprintf("[istag-restore] Looking up reference tag: %s/%s", refNamespace, imageStreamTag.Tag.From.Name))
+				client, err := clients.ImageClient()
+				_, err = client.ImageStreamTags(refNamespace).Get(imageStreamTag.Tag.From.Name, metav1.GetOptions{})
+				if err == nil {
+					p.Log.Info("[istag-restore] reference tag found in cluster")
+				} else {
+					p.Log.Info("[istag-restore] reference tag not found in cluster, adding to restore")
+					addImageStream := false
+					nameSplit := strings.SplitN(imageStreamTag.Tag.From.Name, ":", 2)
+					if len(nameSplit) == 2 {
+						_, err = client.ImageStreams(refNamespace).Get(nameSplit[0], metav1.GetOptions{})
+						if err == nil {
+							p.Log.Info("[istag-restore] reference imagestream found in cluster")
+						} else {
+							p.Log.Info("[istag-restore] reference imagestream not found in cluster, adding to restore")
+							addImageStream = true
+						}
+					}
+
+					additionalItems = []velero.ResourceIdentifier{
 						{
 							GroupResource: schema.GroupResource{
 								Group:    "image.openshift.io",
-								Resource: "imagestreams",
+								Resource: "imagestreamtags",
 							},
 							Namespace: refNamespace,
-							Name:      nameSplit[0],
+							Name:      imageStreamTag.Tag.From.Name,
 						},
-					}, additionalItems...)
+					}
+					if addImageStream {
+						additionalItems = append([]velero.ResourceIdentifier{
+							{
+								GroupResource: schema.GroupResource{
+									Group:    "image.openshift.io",
+									Resource: "imagestreams",
+								},
+								Namespace: refNamespace,
+								Name:      nameSplit[0],
+							},
+						}, additionalItems...)
+					}
 				}
 			}
 		}
-	}
-	// Restore the tag if this is a reference tag *or* an external image. Otherwise,
-	// image import will create the imagestreamtag automatically.
-	if referenceTag || !localImage {
-		p.Log.Info("[istag-restore] Restoring reference or remote imagestreamtag")
-		return &velero.RestoreItemActionExecuteOutput{
-			UpdatedItem:     input.Item,
-			AdditionalItems: additionalItems,
-		}, nil
+
+		// Restore the tag if this is a reference tag *or* an external image. Otherwise,
+		// image import will create the imagestreamtag automatically.
+		if referenceTag || !localImage {
+			p.Log.Info("[istag-restore] Restoring reference or remote imagestreamtag")
+			return &velero.RestoreItemActionExecuteOutput{
+				UpdatedItem:     input.Item,
+				AdditionalItems: additionalItems,
+			}, nil
+		} else {
+			p.Log.Info("[istag-restore] Not restoring local imagestreamtag")
+			return velero.NewRestoreItemActionExecuteOutput(input.Item).WithoutRestore(), nil
+		}
 	} else {
-		p.Log.Info("[istag-restore] Not restoring local imagestreamtag")
-		return velero.NewRestoreItemActionExecuteOutput(input.Item).WithoutRestore(), nil
+		p.Log.Info("[istag-restore] Restoring imagestreamtag")
+		return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
```

Comment:
This seems like the right thing to do in the "no migration annotations found" case -- basically returning the item unchanged as if the plugin isn't running.